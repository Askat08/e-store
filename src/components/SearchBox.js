import React from "react";

export default function SearchBox({
  onSearchClick,
  onSearchChange,
  searchField,
}) {
  return (
    <div className="input-group col-sm-9 mx-auto">
      <input
        className="form-control py-2 shadow-none"
        type="search"
        value={searchField}
        placeholder="Search..."
        id="example-search-input"
        onChange={onSearchChange}
        onKeyPress={onSearchClick}
      />
      <span className="input-group-append">
        <button
          className="btn btn-outline-info shadow-none"
          type="button"
          onClick={onSearchClick}
        >
          <i className="fa fa-search"></i>
        </button>
      </span>
    </div>
  );
}
