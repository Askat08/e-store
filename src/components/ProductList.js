import React from "react";
import Product from "./Product";
import SearchBox from "./SearchBox";

const ProductList = (props) => {
  const {
    items,
    handleAddtoCart,
    messageAlert,
    isError,
    onSearchChange,
    onSearchClick,
    searchfield,
  } = props;

  const comics = items
    .map((comic, i) => {
      return (
        <Product
          key={comic.id}
          id={comic.id}
          price={comic.prices[0].price}
          title={comic.title}
          path={comic.thumbnail.path}
          ext={comic.thumbnail.extension}
          description={comic.description}
          handleAddtoCart={handleAddtoCart}
          messageAlert={messageAlert}
        />
      );
    })
    .slice(0, 15); // render only first 15 items

  // checking if items exists to render and no error

  if (items.length === 0 && !isError) {
    return (
      <div className="d-flex align-items-center justify-content-center">
        <div className="">
          <SearchBox
            onSearchChange={onSearchChange}
            searchField={searchfield}
            onSearchClick={onSearchClick}
          />
        </div>
        <strong>Loading...</strong>
        <div
          className="spinner-border m-5"
          role="status"
          aria-hidden="true"
        ></div>
      </div>
    );
  } else if (isError) {
    return (
      <div>
        <p className="lead text-center">Data not found</p>
        <SearchBox onSearchChange={onSearchChange} searchField={searchfield} />
      </div>
    );
  } else {
    return (
      <React.Fragment>
        <div className="jumbotron m-0">
          <div className="">
            <SearchBox
              onSearchChange={onSearchChange}
              searchField={searchfield}
              onSearchClick={onSearchClick}
            />
          </div>
          <div className="row m-5 justify-content-center">{comics}</div>
        </div>
      </React.Fragment>
    );
  }
};

export default ProductList;
