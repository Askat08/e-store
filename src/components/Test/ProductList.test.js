import React from "react";
import { shallow, mount } from "enzyme";
import ProductList from "../ProductList";
import SearchBox from "../SearchBox";

const props = {
  items: [],
  handleAddtoCart: () => true,
  messageAlert: false,
  isError: false,
  onSearchChange: () => true,
  onSearchClick: () => true,
  searchfield: "",
};

describe("ProductList component test", () => {
  it("should render ProductList component", () => {
    const wrapper = mount(<ProductList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have proper props for div", () => {
    const wrapper = shallow(<ProductList {...props} />);
    expect(wrapper.find("div").at(0).props().className).toEqual(
      "d-flex align-items-center justify-content-center"
    );

    expect(
      wrapper
        .find("div")
        .at(1)
        .containsMatchingElement(
          <SearchBox
            onSearchChange={props.onSearchChange}
            searchField={props.searchfield}
            onSearchClick={props.onSearchClick}
          />
        )
    ).toBeTruthy();

    expect(wrapper.find("div").at(2).props()).toEqual({
      className: "spinner-border m-5",
      role: "status",
      "aria-hidden": "true",
    });
  });

  it("should have proper children for div", () => {
    const wrapper = shallow(<ProductList {...props} />);
    expect(wrapper.find("div").at(0).childAt(0).type()).toEqual("div");
    expect(wrapper.find("div").at(0).childAt(1).type()).toEqual("strong");
    expect(wrapper.find("div").at(0).childAt(2).type()).toEqual("div");
  });
});
