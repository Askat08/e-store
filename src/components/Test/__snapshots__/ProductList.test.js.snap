// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList component test should render ProductList component 1`] = `
<ProductList
  handleAddtoCart={[Function]}
  isError={false}
  items={Array []}
  messageAlert={false}
  onSearchChange={[Function]}
  onSearchClick={[Function]}
  searchfield=""
>
  <div
    className="d-flex align-items-center justify-content-center"
  >
    <div
      className=""
    >
      <SearchBox
        onSearchChange={[Function]}
        onSearchClick={[Function]}
        searchField=""
      >
        <div
          className="input-group col-sm-9 mx-auto"
        >
          <input
            className="form-control py-2 shadow-none"
            id="example-search-input"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search comic book..."
            type="search"
            value=""
          />
          <span
            className="input-group-append"
          >
            <button
              className="btn btn-outline-info shadow-none"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-search"
              />
            </button>
          </span>
        </div>
      </SearchBox>
    </div>
    <strong>
      Loading...
    </strong>
    <div
      aria-hidden="true"
      className="spinner-border m-5"
      role="status"
    />
  </div>
</ProductList>
`;
